client/src/components/Form/Elements/FormData/FormData.vue

```diff
@@ -0,0 +1,508 @@
+<script setup lang="ts">
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faCopy, faFile, faFolder } from "@fortawesome/free-regular-svg-icons";
+import { faExclamation, faLink, faUnlink } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { BButton, BButtonGroup, BFormCheckbox } from "bootstrap-vue";
+import Vue, { computed, onMounted, type Ref, ref, watch } from "vue";
+
+import { getGalaxyInstance } from "@/app";
+import { type EventData, useEventStore } from "@/stores/eventStore";
+
+import type { DataOption } from "./types";
+import { BATCH, SOURCE, VARIANTS } from "./variants";
+
+import FormSelect from "@/components/Form/Elements/FormSelect.vue";
+
+library.add(faCopy, faExclamation, faFile, faFolder, faLink, faUnlink);
+
+interface SelectOption {
+    label: string;
+    value: DataOption | null;
+}
+
+const props = withDefaults(
+    defineProps<{
+        multiple?: boolean;
+        optional?: boolean;
+        options: Record<string, Array<DataOption>>;
+        value?: {
+            values: Array<DataOption>;
+        };
+        extensions?: Array<string>;
+        type?: string;
+        flavor?: string;
+        tag?: string;
+    }>(),
+    {
+        multiple: false,
+        optional: false,
+        value: null,
+        extensions: () => [],
+        type: "data",
+        flavor: null,
+        tag: null,
+    }
+);
+
+const eventBus = new Vue();
+
+const eventStore = useEventStore();
+
+const $emit = defineEmits(["input"]);
+
+// Determines wether values should be processed as linked or unlinked
+const currentLinked = ref(true);
+
+// Indicates which of the select field from the set of variants is currently shown
+const currentField = ref(0);
+
+// Field highlighting status
+const currentHighlighting: Ref<string | null> = ref(null);
+
+// Drag/Drop related values
+const dragData: Ref<EventData | null> = ref(null);
+const dragTarget: Ref<EventTarget | null> = ref(null);
+
+/** Store options which need to be preserved **/
+const keepOptions: Record<string, SelectOption> = {};
+
+/** Displays a wait indicator between setting a value and receiving an update */
+const waiting = ref(false);
+
+/**
+ * Determine whether the file dialog can be used or not
+ */
+const canBrowse = computed(() => variant.value && !!variant.value.find((v) => v.src === SOURCE.DATASET));
+
+/**
+ * Provides the currently shown source type
+ */
+const currentSource = computed(() => (currentVariant.value ? currentVariant.value.src : null));
+
+/**
+ * Interface between incoming input value and select field value
+ */
+const currentValue = computed({
+    get: () => {
+        eventBus.$emit("waiting", false);
+        return getValue();
```

Comment:
The use of an event bus here looks like a somewhat dubious shortcut, as does the somewhat buried passing of the props inside getValue. maybe it's fine though, it'd be good to have @galaxyproject/wg-uiux look at this.

Reply:
I removed the eventbus use case here, and instead pass a regular prop from the parent component to show/hide the spinner. I also removed `getValue` and instead directly inlined its code to avoid confusion.